src/Microsoft.Extensions.Caching.Memory/MemoryCache.cs

```diff
@@ -43,8 +44,8 @@ public MemoryCache(IOptions<MemoryCacheOptions> optionsAccessor)
             }
 
             var options = optionsAccessor.Value;
-            _entries = new Dictionary<object, CacheEntry>();
-            _entryLock = new ReaderWriterLockSlim();
+            _entries = new ConcurrentDictionary<object, CacheEntry>();
```

Comment:
Is this for perf or readability?  If it's just the latter, personally I find two fields pointing to the same object more confusing than doing the cast when needed.  Do we do this anywhere else in our codebase?